packages/lesswrong/components/bookmarks/BookmarksPage.tsx

```diff
@@ -0,0 +1,99 @@
+import { registerComponent, Components } from '../../lib/vulcan-lib';
+import React, { useEffect, useState } from 'react';
+import withErrorBoundary from '../common/withErrorBoundary';
+import {AnalyticsContext} from "../../lib/analyticsEvents";
+import {useCurrentUser} from "../common/withUser"
+import Tabs from '@material-ui/core/Tabs';
+import Tab from '@material-ui/core/Tab';
+import { isEAForum } from '../../lib/instanceSettings';
+import { preferredHeadingCase } from '../../lib/forumTypeUtils';
+import { useLocation, useNavigation } from '../../lib/routeUtil';
+
+type TabType = 'bookmarks' | 'readhistory' | 'votehistory';
+
+const styles = (theme: ThemeType): JssStyles => ({
+  headline: {
+    color: theme.palette.grey[1000],
+    fontSize: isEAForum ? 28 : undefined,
+    fontFamily: isEAForum ? undefined : theme.palette.fonts.serifStack,
+    marginTop: isEAForum ? 10 : 0,
+    marginBottom: 20,
+    [theme.breakpoints.down('sm')]: {
+      marginTop: 20,
+      marginBottom: 10,
+    }
+  },
+  tabs: {
+    marginBottom: 20,
+  },
+  tab: {
+    fontSize: 14,
+    fontWeight: isEAForum ? '700' : undefined,
+    [theme.breakpoints.down('xs')]: {
+      fontSize: 13,
+    }
+  }
+});
+
+const BookmarksPage = ({ classes }: {
+  classes: ClassesType
+}) => {
+  const { history } = useNavigation()
+  const { location } = useLocation()
+  const [activeTab, setActiveTab] = useState<TabType>('bookmarks')
+  
+  useEffect(() => {
+    // unfortunately the hash is unavailable on the server, so we check it here instead
+    if (location.hash === '#readhistory') {
+      setActiveTab('readhistory')
+    } else if (location.hash === '#votehistory') {
+      setActiveTab('votehistory')
+    }
+    //No exhaustive deps because this is supposed to run only on mount
```

Comment:
I'm not convinced by this - if I'm already on the page and I manually change the hash from "readhistory" to "votehistory", I'd expect the tab to change